src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -409,36 +412,48 @@ public List<Course> getAllCoursesToRegister() {
      *
      * @param courseId the courseId for which exercises, lectures, exams and learning goals should be fetched
      * @param refresh  if true, this request was initiated by the user clicking on a refresh button
-     * @return a course with all exercises, lectures, exams, learning goals, etc. visible to the user
+     * @return a course with all exercises, lectures, exams, learning goals, etc. visible to the user. In addition, scores per exercise type for each exercise are sent back as an
+     *         optimization.
      */
     // TODO: we should rename this into courses/{courseId}/details
     @GetMapping("courses/{courseId}/for-dashboard")
     @PreAuthorize("hasRole('USER')")
-    public Course getCourseForDashboard(@PathVariable long courseId, @RequestParam(defaultValue = "false") boolean refresh) {
+    public CourseForDashboardDTO getCourseForDashboard(@PathVariable long courseId, @RequestParam(defaultValue = "false") boolean refresh) {
         long timeNanoStart = System.nanoTime();
         log.debug("REST request to get one course {} with exams, lectures, exercises, participations, submissions and results, etc.", courseId);
         User user = userRepository.getUserWithGroupsAndAuthorities();
         Course course = courseService.findOneWithExercisesAndLecturesAndExamsAndLearningGoalsAndTutorialGroupsForUser(courseId, user, refresh);
         courseService.fetchParticipationsWithSubmissionsAndResultsForCourses(List.of(course), user);
```

Comment:
(minor) this method calls `List.of()` 3 times, I think this should be a variable.